### PR TITLE
Updated return value information for Marshal.GetExceptionForHR

### LINKED ISSUE #1950: Describe circumstances in which this function will return null
For instance, it can return null if you pass 0 for the hresult. Are there any other cases?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7979e2b7-2226-efdd-456f-4300b222d898
* Version Independent ID: 8cb33d7f-a658-45ee-a2fc-013519feeb88
* Content: [Marshal.GetExceptionForHR Method (System.Runtime.InteropServices)](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.getexceptionforhr?view=netframework-4.7.2)
* Content Source: [xml/System.Runtime.InteropServices/Marshal.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Runtime.InteropServices/Marshal.xml)
* Product: **dotnet-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by rpetrusha**
## Updated return value information for Marshal.GetExceptionForHR

Fixes #1950

### CODE REVIEW

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 1]
```diff
@@ -3362,7 +3362,7 @@
       <Docs>
         <param name="errorCode">The HRESULT to be converted.</param>
         <summary>Converts the specified HRESULT error code to a corresponding <see cref="T:System.Exception" /> object.</summary>
-        <returns>An object that represents the converted HRESULT.</returns>
+        <returns>An object that represents the converted HRESULT, or <see langword="null" /> if an HRESULT value, such as <see langword="S_OK" /> or <see langword="S_FALSE" />, does not represent an error code.</returns>
```
Comment (mairaw) [inline xml/System.Runtime.InteropServices/Marshal.xml]: Wondering if this flow would work better

```suggestion
        <returns>An object that represents the converted HRESULT, or <see langword="null" /> if the HRESULT value doesn't represent an error code (for example, <see langword="S_OK" /> or <see langword="S_FALSE" />) .</returns>
```

Comment (rpetrusha) [reply xml/System.Runtime.InteropServices/Marshal.xml]: This does work much better, @mairaw. Thanks.

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 2]
```diff
@@ -3362,7 +3362,7 @@
       <Docs>
         <param name="errorCode">The HRESULT to be converted.</param>
         <summary>Converts the specified HRESULT error code to a corresponding <see cref="T:System.Exception" /> object.</summary>
-        <returns>An object that represents the converted HRESULT.</returns>
+        <returns>An object that represents the converted HRESULT, or <see langword="null" /> if the HRESULT value doesn't represent an error code (for example, <see langword="S_OK" /> or <see langword="S_FALSE" />) .</returns>
```
Comment (mairaw) [inline xml/System.Runtime.InteropServices/Marshal.xml]:
```suggestion
        <returns>An object that represents the converted HRESULT, or <see langword="null" /> if the HRESULT value doesn't represent an error code (for example, <see langword="S_OK" /> or <see langword="S_FALSE" />).</returns>
```

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 3]
```diff
@@ -3425,7 +3425,7 @@
         <param name="errorCode">The HRESULT to be converted.</param>
         <param name="errorInfo">A pointer to the <see langword="IErrorInfo" /> interface that provides more information about the error. You can specify <c>IntPtr(0)</c> to use the current <see langword="IErrorInfo" /> interface, or <c>IntPtr(-1)</c> to ignore the current <see langword="IErrorInfo" /> interface and construct the exception just from the error code.</param>
         <summary>Converts the specified HRESULT error code to a corresponding <see cref="T:System.Exception" /> object, with additional error information passed in an [IErrorInfo](https://msdn.microsoft.com/library/4dda6909-2d9a-4727-ae0c-b5f90dcfa447) interface for the exception object.</summary>
-        <returns>An object that represents the converted HRESULT and information obtained from <paramref name="errorInfo" />.</returns>
+        <returns>An object that represents the converted HRESULT and information obtained from <paramref name="errorInfo" />, or <see langword="null" /> if an HRESULT value, such as <see langword="S_OK" /> or <see langword="S_FALSE" />, does not represent an error code.</returns>
```
Comment (mairaw) [inline xml/System.Runtime.InteropServices/Marshal.xml]:
```suggestion
        <returns>An object that represents the converted HRESULT and information obtained from <paramref name="errorInfo" />, or <see langword="null" /> if an HRESULT value doesn't represent an error code (for example, <see langword="S_OK" /> or <see langword="S_FALSE" />).</returns>
```

**Pull request merged**